extensions/PhabBugz/lib/Util.pm

```diff
@@ -382,4 +387,46 @@ sub request {
     return $result;
 }
 
+sub get_security_sync_groups {
+    my $bug = shift;
+
+    my $phab_sync_groups = Bugzilla->params->{phabricator_sync_groups}
+        || ThrowUserError('invalid_phabricator_sync_groups');
+    my $sync_group_names = [ split('[,\s]+', $phab_sync_groups) ];
+
+    my $bug_groups = $bug->groups_in;
+    my $bug_group_names = [ map { $_->name } @$bug_groups ];
+
+    my @set_groups = intersect($bug_group_names, $sync_group_names);
+
+    return @set_groups;
+}
+
+sub add_security_sync_comments {
+    my ($revisions, $bug) = @_;
+
+    my $phab_error_message = 'Revision is being made private due to unknown Bugzilla groups.';
+
+    foreach my $revision (@$revisions) {
+        add_comment_to_revision( $revision->{phid}, $phab_error_message );
+    }
+
+    my $num_revisions = 0 + @$revisions;
+    my $bmo_error_message =
+    ( $num_revisions > 1
+    ? $num_revisions.' revisions were'
+    : 'One revision was' )
+    . ' made private due to unknown Bugzilla groups.';
+
+    my $user = Bugzilla::User->new( { name => PHAB_AUTOMATION_USER } );
```

Comment:
You could even not do this if the user is already PHAB_AUTOMATION_USER in the case of an API call and not the push connector:

if (Bugzilla->user->name ne PHAB_AUTOMATION_USER) {

}